.github/workflows/hardware-long.yml

```
@@ -136,69 +137,97 @@ jobs:
             --tag e2e \
             --hardware-map /opt/tenstorrent/twister/hw-map.yml --west-flash \
             -T ../tt-zephyr-platforms/app -ll DEBUG
-
-
-      - name: Fix APT URLs
+      - name: Install TT-SMI and reset the device
         run: |
-          # Patch APT URLs to use https
           sed -i -e 's/http/https/g' /etc/apt/sources.list
           apt update
-
-      - name: Install clang 17
-        run: |
-          wget https://apt.llvm.org/llvm.sh
-          sed -i "s/http:\/\/apt.llvm.org/https:\/\/apt.llvm.org/" -i llvm.sh
-          chmod +x llvm.sh
-          ./llvm.sh 17 all
-          rm -f llvm.sh
-
+          apt install -y cargo
+          pip install git+https://github.com/tenstorrent/tt-smi@v3.0.5
+          tt-smi -r 0
+  hardware-metal-test:
+    needs:
+      - hardware-zephyr-test
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - board: p100
+            runs-on:
+              - yyz-zephyr-lab-p100
+          - board: p100a
+            runs-on:
+              - yyz-zephyr-lab-p100a
+    runs-on: ${{ matrix.config.runs-on }}
+    container:
+      image: ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-dev-amd64:latest
+      volumes:
+        - /dev/hugepages-1G:/dev/hugepages-1G
+        - /dev/hugepages:/dev/hugepages
+      options: '--device /dev/tenstorrent --device /dev/bus/usb --privileged'
+    env:
+      ARCH_NAME: blackhole
+      TT_METAL_HOME: ${{ github.workspace }}/tt-metal
+      PYTHONPATH: ${{ github.workspace }}/tt-metal
+    steps:
       - name: Install Git LFS
         run: |
+          apt update
           apt install -y git-lfs
-
+      - name: Checkout Zephyr Platforms (only for rescan-pcie.sh)
+        uses: actions/checkout@v4
+        with:
+          path: tt-zephyr-platforms
+          ref: ${{ github.ref }}
+      - name: Run the rescan-pcie.sh script
+        run: |
+          ./tt-zephyr-platforms/scripts/rescan-pcie.sh
```

It's possible that flashing via `tt-flash` will flash + rescan pcie here. I vaguely recall that someone said something about `tt-smi -r` and `tt-flash` doing that.

In any case, it seems unnecessary to check out tt-zephyr-platforms for 1 script.

Maybe just this?
```suggestion
      - name: Rescan PCIe
        run: |
          curl -o /tmp/rescan-pcie.sh \
          'https://github.com/tenstorrent/tt-zephyr-platforms/blob/cc07066/scripts/rescan-pcie.sh'
           /tmp/rescan-pcie.sh
```

---

tt-flash and tt-smi -r did not do that in my testing on the lab machine. I am happy to be wrong, but for now only rescanning works.

---

> I like this! Maybe I would just use `git_ref` to get the current version of the script :)

There is a git ref in the link:

> https://github.com/tenstorrent/tt-zephyr-platforms/blob/cc07066/scripts/rescan-pcie.sh

---

Addressed.